clientapi/routing/leaveroom.go

```diff
@@ -41,7 +41,7 @@ func LeaveRoomByID(
 		if leaveRes.Code != 0 {
 			return util.JSONResponse{
 				Code: leaveRes.Code,
-				JSON: jsonerror.Forbidden(leaveRes.Message),
+				JSON: jsonerror.LeaveServerNoticeError(),
```

Comment:
There's no guarantee that this is because of server notices though?

Reply:
The `PerformLeaveResponse` struct just got the `Code` in [2e1e0c81a19830cbf2055a6d15c5d7424c2d0087](https://github.com/matrix-org/dendrite/pull/2180/commits/2e1e0c81a19830cbf2055a6d15c5d7424c2d0087#diff-7c6dd331dc0262e2eb97b598ef5d4a8cbb70050a4291249d05af21dfc26e0086), so right now it should be guaranteed.

Reply:
That's not very extensible or kind to the next person who wants to add an additional code :) it would be nicer to just proxy back the JSON blob to send back so instead of `Message string` having `Message interface{}` which is JSON serialisable should work.